src/Microsoft.AspNet.Mvc.Razor/RazorViewEngine.cs

```diff
@@ -228,6 +228,29 @@ public virtual IEnumerable<string> AreaViewLocationFormats
             return new RazorPageResult(pageName, searchedLocations);
         }
 
+        private string GetNormalizedRouteValue(string key, ActionContext context)
+        {
+            if (context.ActionDescriptor.AttributeRouteInfo == null)
+            {
+                var match = context.ActionDescriptor.RouteConstraints.SingleOrDefault(r => r.RouteKey == key);
```

Comment:
For safety we want to do a case-insensitive comparison here. `RouteValueDefaults` is already case-insensitive.